search/filters/ExactMatchFilter.php

```diff
@@ -50,6 +50,7 @@ protected function applyOne(DataQuery $query) {
 	 */
 	protected function applyMany(DataQuery $query) {
 		$this->model = $query->applyRelation($this->relation);
+		$column = $query->lastAlias ? sprintf('"%s"."%s"', $query->lastAlias, $this->name) : $this->getDbName();
```

Comment:
I think lastAlias should be returned from applyRelation, since it's specific to that method call, not necessarily to that object, so it shouldn't be a member.

It could either be a part of the return value (array of data including class), or a by-ref parameter.